src/main/scala/org/scalacheck/Arbitrary.scala

```diff
@@ -152,8 +152,8 @@ private[scalacheck] sealed trait ArbitraryLowPriority {
   implicit lazy val arbChar: Arbitrary[Char] = Arbitrary {
     // valid ranges are [0x0000, 0xD7FF] and [0xE000, 0xFFFD].
     //
-    // ((0xFFFD + 1) - 0xE000) + ((0xD7FF + 1) - 0x0000)
-    choose(0, 63486).map { i =>
+    // ((0xFFFD + 1) - 0xE000) + ((0xD7FF + 1) - 0x0000) - 1
+    choose(0, 63485).map { i =>
```

Comment:
So it was just a miscalculation of the range that was missed, or was `Gen.choose` not inclusive at some point?

Reply:
I think I just need to spend time to grok the math.

Reply:
Oh, so zero-based iteration from 0 to i needs to be exclusive upper bound, but `Gen.choose` is inclusive bounds.  The range is actually 63486 values, but we're computing from zero.

Reply:
By example, the ranges of [0, 2] and [6, 8] are actually a total of 6 values:

(_2_ - _0_ + _1_) + (_8_ - _6_ + _1_) = _3_ + _3_ = _6_

However, zero-based iteration through the range of values should be upper bound exclusive, [0, 6).

```
scala> (0 to 5).map { i =>                                                      
     |   if (i <= 2) i
     |   else i + 3
     | }
     |
val res0: IndexedSeq[Int] = Vector(0, 1, 2, 6, 7, 8)
```

Reply:
Yes, that's the cause.
I wrote almost the same code during development. 😄